micro_http/src/request.rs

```diff
@@ -152,19 +152,25 @@ impl<'a> Request<'a> {
     /// ```
     pub fn try_from(byte_stream: &'a [u8]) -> Result<Self, RequestError> {
         // The first line of the request is the Request Line. The line ending is LF.
-        let (request_line, _) = split(byte_stream, LF);
+        let (request_line, rest) = split(byte_stream, LF);
```

Comment:
Let's follow the HTTP protocol when splitting this.
Specifically when splitting a request by `LF` you end up with a `request_line` and another byte stream representing the `headers_and_body`.

To get all the headers we need a new split by CRLF. Only after you parsed both the request line and headers you proceed in creating a Request object.

```rust
    pub fn try_from(byte_stream: &'a [u8]) -> Result<Self, RequestError> {
        // The first line of the request is the Request Line. The line ending is LF.
        let (request_line, headers_and_body) = split(byte_stream, LF);
        if request_line.len() < RequestLine::min_len() {
            return Err(RequestError::InvalidRequest);
        }

        // The Request Line should include the trailing LF.
        let request_line = RequestLine::try_from(&byte_stream[..=request_line.len()])?;

        // Now let's split `headers_and_body` by LF to get only the headers so we can parse them.
        // We don't care about the body for now.
        // The split function needs to be updated so that you can split by CRLF. This is not yet
        // supported as far as I remember.
        let (headers, _) = split(headers_and_body, LF);
        let headers = Headers::try_from(headers)?;

         Ok(Request {
            request_line,
            headers,
            body: None,
        })
    }
```

Reply:
As per our offline conversation this code will be changed anyway because we need to support streaming which will be implementing as part of the RequestConnection work.